### DISCUSSION TITLE
What does "legality" mean now that we the players control WFC?

### DISCUSSION THREAD

**Discussion by ketsuban** [mm201/pkmnFoundations#104] [Ideas]
The roadmap document talks about legality checks and "the features of Pokecheck", but it seems like those features only fully made sense in their original context where the servers were controlled by Nintendo and what was or was not legal was at their whim.

Take hidden abilities, for example. [Several](https://tcrf.net/Pokémon_Black_and_White#Unused_Hidden_Abilities) were never made available (or were only distributed on male Pokémon) so they're "illegal". However, there's no technical reason for this; it's just what Nintendo happened to decide to distribute and how. Will our Game Sync infrastructure prevent people transferring female Litwick into the Entree Forest in order to preserve this "illegality"?

Then there's the Mystery Gift. Currently it's just cycling through Project Pokémon's archive of past events, one per day; the roadmap mentions a monthly rotation (presumably something like [the one suggested here](https://gbatemp.net/threads/save-nintendo-wifi-a-project-to-save-online-servers-for-ds-and-wii-games.362717/page-31#post-5004623)). However, there were some things that were never distributed in Gen IV; with what gets distributed over Mystery Gift now up to us, there's no reason for these to remain inaccessible.

* The Member Card and Oak's Letter in Diamond and Pearl.
* The Azure Flute and Lock Capsule. The Azure Flute allows access to the Sinjoh Ruins in HeartGold/SoulSilver, which has some different dialogue if you already visited it once (with an event Arceus); the Lock Capsule allows access to TM95 Snarl in Black and White.
* Miscellany: The Comet accessory, a backdrop, the Stopwatch and Alarm Clock Poketch apps, and (outside Japanese Diamond and Pearl) the Lucario Doll decoration.
* Head Smash Nosepass in HeartGold/SoulSilver. (Yes, they added an unobtainable egg move and then removed it in Gen V.)

I'm not even going to get into shiny-locked Pokémon; of note is the recent distribution of a shiny Arceus, which never happened in Gen IV or V.

**Comment by mm201** [comment]
I only intend on blocking things that would give cheaters an unfair advantage in PvP, so illegal movesets, EVs above 510, and, yes, unobtainable hidden abilities count here, so Shadow Tag Litwick is out. I also want to block from the GTS anything that wouldn't make it through Pokebank, so that people don't get stiffed with hacked Pokemon they can't take with them.

I have no plans on doing 'forensic' style checks like validating RNG output, trash bytes, or encounter locations. Hackers can get around these checks anyway.

**Comment by mm201** [comment]
Since this is more of a question than an actual issue, and there'a already an [open issue](https://github.com/mm201/pkmn-classic-framework/issues/3) for cheat detection, I'll go ahead and close this.

**Comment by ketsuban** [comment]
Hi! I'm coming back to this after four years. My brain is a mystery.

> I only intend on blocking things that would give cheaters an unfair advantage in PvP, so illegal movesets, EVs above 510, and, yes, unobtainable hidden abilities count here, so Shadow Tag Litwick is out.

I'll accept the first two examples as unambiguous, but "unobtainable" hidden abilities were precisely one of the things I was asking about.

A level 5 Dragonite, a Pokemon with 1,530 EVs, a Wonder Guard Spiritomb or a Sandy Wormadam with Leaf Storm are all clearly illegal because the game encodes a set of rules which prevent those coming into existence, but the nonexistence of Shadow Tag Litwick is a consequence of the arbitrary nondistribution of Litwick in the Dream World. The game says that in the event a Litwick with its hidden ability is created that ability would be Shadow Tag, but defines no rules for when a Litwick with its hidden ability is created—that responsibility is delegated to the Dream World servers, which at the time were controlled entirely by Nintendo. They're now controlled by you (and more broadly by the player base, assuming you continue to be willing to entertain other peoples' ideas) so in what sense are Shadow Tag Litwicks generated by your Legally Noninfringing Hidden Ability Distribution Engine illegitimate? Why are Nintendo's arbitrary decisions in 2012 beyond question?

I hope I'm not coming across as too pushy here - I genuinely do just want to understand your rationale, and wonder whether you're placing too much emphasis on arbitrary decisions from a decade ago by a company that doesn't care outside of how much money they can make.

**Comment by mm201** [comment]
So if I understand you correctly, you're asking that *since we now control the Game Sync and Mystery Gift servers, why can't we just make new events?* Your question isn't really about legality checks. It's a valid point and I do have interest in providing certain things like the unreleased Poketch apps and C-gear skins. We would need to approach it on a case by case basis. For instance, Game Freak has *retconned away* Shadow Tag Litwick and instead given it Infiltrator. Shadow Tag Litwick doesn't make it through Pokebank, so I don't want to give it to anyone. Some things like HA female starters/Gothita are in kind of a grey area. Obviously PkHex will flag them but they don't provide a competitive advantage, are released on later gens, and probably won't get blocked.

Certainly I think we can agree that events my server is hosting should be considered legitimate by its checks. That's only logical. But PCN is part of a larger community, and I don't think I should be making these kinds of decisions on my own. We can talk more once I actually get Game Sync and Mystery Gift implemented.

**Comment by ketsuban** [comment]
Thanks for replying.

> For instance, Game Freak has retconned away Shadow Tag Litwick and instead given it Infiltrator. Shadow Tag Litwick doesn't make it through Pokebank, so I don't want to give it to anyone.

That's a reasonable point. I was starting with the assumption that anyone who wanted to transfer stuff from a Gen 5 game would already have done so and that we're essentially an isolated ecosystem as a result, but I can imagine someone trying it anyway regardless of whether it's a worthwhile thing to do.

That said, can you provide evidence that Bank rejects Shadow Tag Litwick? I found [a post claiming that Bank will silently update its ability to Infiltrator and accept it](https://pokemondb.net/pokebase/196250/will-litwicks-ha-transfer-to-x-y), but obviously one person can be wrong on the internet as easily as another.

> Certainly I think we can agree that events my server is hosting should be considered legitimate by its checks. That's only logical. But PCN is part of a larger community, and I don't think I should be making these kinds of decisions on my own. We can talk more once I actually get Game Sync and Mystery Gift implemented.

Absolutely. I look forward to finding out what gets set up to feel the pulse of the community.
